packages/gcf-utils/src/logging/gcf-logger.ts

```diff
@@ -104,14 +105,29 @@ export class GCFLogger {
   /**
    * Log at the metric level
    */
+  public metric(msg: string, entry: LogEntry | string): void;
   public metric(msg: string, ...args: any[]): void;
   public metric(obj: object, msg?: string, ...args: any[]): void;
   public metric(
     objOrMsg: object | string,
-    addMsg?: string,
+    addMsg?: LogEntry | string,
     ...args: any[]
   ): void {
-    this.log('metric', objOrMsg, addMsg, ...args);
+    let payload: LogEntry = {
+      count: 1,
+      event: 'unknown',
+      type: 'metric',
+    };
+    if (typeof objOrMsg === 'string') {
+      payload.event = objOrMsg;
+    } else {
+      payload = {...payload, ...objOrMsg};
+    }
+    if (typeof addMsg === 'object') {
+      payload = {...payload, ...addMsg};
+      addMsg = undefined;
+    }
+    this.log('metric', payload, addMsg, ...args);
```

Comment:
Is it ok if `objOrMsg` or `addMsg` overwrites `{type: metric}`? If not, perhaps `{...payload, type: 'metric'}` could be used here

Reply:
I was on the fence about whether we should allow an upstream application to override `type`.

I think you're correct that we probably shouldn't allow this, since someone might accidentally set this field, and then not have the metrics data in our BigQuery sink. I've updated the approach and added a test.